processes/loc.py

```diff
@@ -0,0 +1,208 @@
+import json
+import os, requests
+from requests.exceptions import HTTPError, ConnectionError
+
+from .core import CoreProcess
+from mappings.core import MappingError
+from mappings.loc import LOCMapping
+from managers import WebpubManifest
+from logger import createLog
+
+logger = createLog(__name__)
+
+LOC_ROOT_OPEN_ACCESS = 'https://www.loc.gov/collections/open-access-books/?fo=json&fa=access-restricted%3Afalse&c=25&at=results'
+LOC_ROOT_DIGIT = 'https://www.loc.gov/collections/selected-digitized-books/?fo=json&fa=access-restricted%3Afalse&c=25&at=results' 
+
+class LOCProcess(CoreProcess):
+
+    def __init__(self, *args):
+        super(LOCProcess, self).__init__(*args[:4])
+
+        self.ingestOffset = int(args[5] or 0)
+        self.ingestLimit = (int(args[4]) + self.ingestOffset) if args[4] else 5000
+        self.fullImport = self.process == 'complete' 
+
+        # Connect to database
+        self.generateEngine()
+        self.createSession()
+
+        # Connect to epub processing queue
+        # self.fileQueue = os.environ['FILE_QUEUE']
+        # self.fileRoute = os.environ['FILE_ROUTING_KEY']
+        # self.createRabbitConnection()
+        # self.createOrConnectQueue(self.fileQueue, self.fileRoute)
+
+        # S3 Configuration
+        self.s3Bucket = os.environ['FILE_BUCKET']
+        self.createS3Client()
+
+    def runProcess(self):
+        count = 0
+        count2 = 0
+        sp = 1
+        openAccessURL = '{}&sp={}'.format(LOC_ROOT_OPEN_ACCESS, sp)
+        try:
+            while sp > 0:
+                openAccessURL = '{}&sp={}'.format(LOC_ROOT_OPEN_ACCESS, sp)
+                jsonData = self.fetchPageJSON(openAccessURL)
+                with open("locAPI_metadata_OPAccess.json", "w", encoding='utf-8') as write_file:
+                    json.dump(jsonData.json(), write_file, ensure_ascii = False, indent = 6)
+                with open("locAPI_metadata_OPAccess.json") as f:
+                    LOCData = json.load(f)
+                    for metaDict in LOCData['results']:
+                        if 'pdf' in metaDict['resources'][0].keys() or 'epub_file' in metaDict['resources'][0].keys():
+                            self.processLOCRecord(metaDict)
+                            count += 1
```

Comment:
question, seems like we're writing to a file, only to read the file and process the records. Is there any need to do any file IO at all? Seems like we could just process the results of `fetchPageJSON` directly

Reply:
You're right and I did it this way because I wanted to keep track of what LOC works that are being processed if an error occurs, so I can know which specific work is causing an error. However, i could simply log out the current url and title of the work every time I run through the process instead to keep track that instead.